File: models/realm.py
import uuid
from services.worldClock import WorldClock

class Realm:
    def __init__(self, name, description="", offset=0):
        self.realmId = uuid.uuid4()
        self.name = name
        self.description = description
        self.offset = offset   

    def toLocalTime(self, worldMins):
        return worldMins + self.offset

    def formatTime(self, worldMins):
        local = self.toLocalTime(worldMins)
        return WorldClock.format(local)

    def __str__(self):
        return self.name

File: INF122Assignment3/src/Realm.java
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
public class Realm {
    private UUID realm_id;
    private String name;
    private String description;
    private int timeOffset; //measured in minutes

    public Realm(){
        this.realm_id = UUID.randomUUID();
        this.name = "Hub";
        this.description = "The Hub of all Realms. QuestEvents rarely occur here, but it is a good place to start your journey or take a break between journeys.";
        this.timeOffset = 0;
    }

    public Realm(String name, String description, int timeOffset) {
        this.realm_id = UUID.randomUUID();
        this.name = name;
        this.description = description;
        this.timeOffset = timeOffset;
    }

    public WorldTime toLocalTime(){
        return new WorldTime(WorldClock.getInstance().now().getMinutes_passed() + timeOffset);
    }

    public WorldTime toLocalTime(int minutes){
        return new WorldTime(minutes + timeOffset);
    }

    @Override
    public String toString(){
        return this.name;
    }
}
